packages/gator-permissions-snap/src/clients/priceApiClient.ts

```diff
@@ -47,9 +54,20 @@ export class PriceApiClient {
       logger.error(
         `HTTP error! Failed to fetch spot price for caipAssetType(${caipAssetType}) and vsCurrency(${vsCurrency}): ${response.status}`,
       );
-      throw new Error(
-        `HTTP error! Failed to fetch spot price for caipAssetType(${caipAssetType}) and vsCurrency(${vsCurrency}): ${response.status}`,
-      );
+
+      if (response.status === 404) {
+        throw new ResourceNotFoundError(
+          `Spot price not found for ${caipAssetType}`,
+        );
+      } else if (response.status >= 500) {
+        throw new ResourceUnavailableError(
+          `Price service temporarily unavailable (HTTP ${response.status})`,
+        );
+      } else {
+        throw new ResourceUnavailableError(
+          `HTTP error ${response.status}: Failed to fetch spot price for ${caipAssetType}`,
+        );
+      }
```

Comment:
Nice work 👍  
Would it make sense to extract the HTTP error mapping into a reusable utility, e.g.  
`packages/gator-permissions-snap/src/utils/httpErrors.ts`?  

That way, the logic stays consistent and can be reused across the codebase:  

```typescript
function mapHttpErrorToSnapError(status: number, context: string): Error {
  if (status === 404) {
    return new ResourceNotFoundError(context);
  } else if (status >= 500) {
    return new ResourceUnavailableError(`Service temporarily unavailable (HTTP ${status})`);
  } else {
    return new ResourceUnavailableError(`HTTP error ${status}: ${context}`);
  }
}
```